### DISCUSSION TITLE
Removing tracks with CLI

### DISCUSSION THREAD

**Discussion by MikeAxtell** [GMOD/jbrowse-components#1457] [Q&A] [answered]
jbrowse2 1.0.0

When using jbrowse CLI tools to manage an instance: Is there a command to delete a track? Or is that accomplished only by hand-editing config.json ?

Sorry if I missed it in the user guide. Couldn't find anything on it in the jbrowse command line help either.

Thanks !

**Comment by cmdcolin** [comment, marked as answer]
That is accurate, hasn't been added yet. So hand editing the config.json is fine for now

But will be added soon! Ref issue https://github.com/GMOD/jbrowse-components/issues/1253

**Reply by zhilianghu** [reply]
I fail to find any updates; is there already an option/tool to remove a track in JBrowse 2?

**Reply by cmdcolin** [reply]
no built in tool yet, so removing by editing the json file manually is probably the current recommended method

**Reply by zhilianghu** [reply]
Thank you for the update.  I notice there is a remove track perl script for JBrowse 1 (https://jbrowse.org/docs/remove-track.pl.html).  (I don't find its source code ...) Is it just to copy over a json.conf for rewrite but only to omit a named structure in order to "remove" it?

**Reply by cmdcolin** [reply]
basically ya, in jb2 you can  (by hand) remove a track entry from the tracks array in a config.json. the jb1 equivalent would do the same from a tracklist.json (the .conf is not supported by that script)

**Reply by zhilianghu** [reply]
On the same line of thoughts - is it possible for the master json file to "include" another json (supposedly some customizable tracks to "add" or "remove" without going through the entire master json config files with hundreds of lines)?  I hope it can be the case where the "include" line in the master json can be comment out or left active easily.

**Reply by zhilianghu** [reply]
As an alternative ... what symbol makes the "comment" lines in config.json?  I tried # , / , // , /* ... */, etc all breaks jbrowse to load.
This is helpful for maintaining large config.json (e.g. delete 20 tracks section from a config.json with 2500 lines).

**Reply by cmdcolin** [reply]
@zhilianghu One thing I would recommend is to not be afraid of manipulating the JSON by hand. JBrowse 1 supplied some features like ignoring comments and including json, but jbrowse 2 is a bit more strict and it is unclear whether we will support it. But in the meantime, you can write helper scripts to accomplish this.

For example, if you want to make a JSON file that includes another JSON file, that can be done externally from jbrowse, and when you are deploying your production server, you run a script to create your final config.json

example, small node.js script

```
const fs = require('fs')
const conf = JSON.parse(fs.readFileSync('config.json'))
for(file in conf.includes) {
   const subconf = JSON.parse(fs.readFileSync(file))
   conf.tracks=conf.tracks.concat(sub.tracks) // or other deepmerge method
}
fs.writeFileSync('config_new.json', JSON.stringify(conf))
```

similar for comments in JSON files: we support only the strict JSON strandard, so no comments. if you want you can write a preparation script to strip out comments before preparing your final config.json file for use in a production instance e.g. `grep -v "#" config_pre.json > config.json`

Note that these are not really related to the issue of removing tracks though, can make new threads for this purpose

**Reply by zhilianghu** [reply]
That's a good idea -- I will make separate pre-config parts with comments then pull them together into the working copy stripping all comments. Perhaps I can make my script interactive for include/exclude based on part names/comments. Thank you!
